src/modules/joinOrderOptimization.js

```diff
@@ -0,0 +1,460 @@
+        switch (op) {
+          case '$eq':
+            if (tableARow[leftExpression] !== tableBRow[rightExpression]) {
+              shouldAddRow = false;
+            }
+            break;
+          case '$neq':
+            if (tableARow[leftExpression] === tableBRow[rightExpression]) {
+              shouldAddRow = false;
+            }
+            break;
+          case '$gt':
+            if (tableARow[leftExpression] <= tableBRow[rightExpression]) {
+              shouldAddRow = false;
+            }
+            break;
+          case '$gte':
+            if (tableARow[leftExpression] < tableBRow[rightExpression]) {
+              shouldAddRow = false;
+            }
+            break;
+          case '$lt':
+            if (tableARow[leftExpression] >= tableBRow[rightExpression]) {
+              shouldAddRow = false;
+            }
+            break;
+          case '$lte':
+            if (tableARow[leftExpression] > tableBRow[rightExpression]) {
+              shouldAddRow = false;
+            }
+            break;
+          default:
+            console.error('Invalid operator');
+        }
+        if (!shouldAddRow) break;
+      }
+      if (shouldAddRow) {
+        const newRow = {...tableARow, ...tableBRow};
+        newRows.push(newRow);
+      }
+    }
+  }
+  return {rows: newRows, cost: totalCost};
+};
+
+/**
+ * Gets the data for the given table
+ * @param tableName - the table name
+ */
+const getTableData = (tableName: string) => {
+  switch (tableName) {
+    case 'Department':
+      return addTablePrefixToData('department', department.data);
+    case 'Doctor':
+      return addTablePrefixToData('doctor', doctor.data);
+    case 'Patient':
+      return addTablePrefixToData('patient', patient.data);
+    default:
+      return [];
+  }
+};
+
+/**
+ * Adds the table prefix to the data
+ * @param tableName - the table name
+ * @param data - the data
+ */
+const addTablePrefixToData = (
+  tableName: string,
+  data: Array<{[key: string]: string}>
+) => {
+  const newData = [];
+  for (const row of data) {
+    const newRow: {[key: string]: string} = {};
+    for (const key of Object.keys(row)) {
+      newRow[`${tableName}.${key}`.toLowerCase()] = row[key];
+    }
+    newData.push(newRow);
+  }
+  return newData;
+};
+
+class JoinOrderQueueElement {
+  graph: Graph;
+  joinTables: Array<string>;
+  rows: Array<any>;
+  cost: number;
+  joinOrder: Array<string | Array<JoinCondition>>;
+
+  constructor(
+    graph: Graph,
+    joinTables: Array<string>,
+    rows: Array<any>,
+    cost: number,
+    joinOrder: Array<string | Array<JoinCondition>>
+  ) {
+    this.graph = graph;
+    this.joinTables = joinTables;
+    this.rows = rows;
+    this.cost = cost;
+    this.joinOrder = joinOrder;
+  }
+
+  getCost = (): number => {
+    return this.cost;
+  };
+
+  getSize = (): number => {
+    return this.joinTables.length;
+  };
+
+  getChildren = (): Array<JoinOrderQueueElement> => {
```

Comment:
Some comments in this function would be helpful to better understand what's going on.